rules/scripts/hsm_init.template.sh

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+# The script must be executed from its local directory.
+usage () {
+  echo "Usage: $0 -i <input.tar.gz> -o <output.tar.gz>"
+  echo "  -m <pkcs.some>     Path to the PKCS#11 module."
+  echo "  -t <token>         Token name."
+  echo "  -s <SoftHSMConfig> Path to the SoftHSM config file."
+  echo "  -p <pin>           PIN for the token."
+  echo "  -i <input.tar.gz>  Path to the input tarball."
+  echo "  -w                 Execute destroy commands before initializing the assets."
+  echo "  -o <output.tar.gz> Path to the output tarball."
+  exit 1
+}
+
+readonly OUTDIR_HSM="hsm"
+readonly OUTDIR_CA="ca"
+readonly OUTDIR_PUB="pub"
+
+readonly INIT_HJSON=@@INIT_HJSON@@
+readonly DESTROY_HJSON=@@DESTROY_HJSON@@
+readonly HSMTOOL_BIN_DEFAULT=@@HSMTOOL_BIN@@
+readonly CERTGEN_TEMPLATES=(@@CERTGEN_TEMPLATES@@)
+readonly CERTGEN_KEYS=(@@CERTGEN_KEYS@@)
+readonly CERTGEN_ENDORSING_KEYS=(@@CERTGEN_ENDORSING_KEYS@@)
+
+HSMTOOL_BIN="${HSMTOOL_BIN:-./${HSMTOOL_BIN_DEFAULT}}"
+
+FLAGS_HSMTOOL_MODULE=""
+FLAGS_HSMTOOL_TOKEN=""
+FLAGS_SOFTHSM_CONFIG=""
+FLAGS_HSMTOOL_PIN=""
+FLAGS_IN_TAR=""
+FLAGS_OUT_TAR=""
+FLAGS_WIPE=false
+
+while getopts 'm:t:s:p:i:o:w' opt; do
+  case "${opt}" in
+    m)
+      FLAGS_HSMTOOL_MODULE="${OPTARG}"
+      ;;
+    t)
+      FLAGS_HSMTOOL_TOKEN="${OPTARG}"
+      ;;
+    s)
+      FLAGS_SOFTHSM_CONFIG="${OPTARG}"
+      ;;
```

Comment:
This flag is optional, correct?  You need it when running in a testing environment with SoftHSM, but otherwise, you do not.

Reply:
Fixe. Made this flag optional. It will only be added to the command if the users sets the `-s` flag.